akka24-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/ControlClient.scala

```diff
@@ -218,6 +238,59 @@ class ControlClient(handler: ConnectionHandler, conductrAddress: URL, apiVersion
   }
 
   /**
+   * Load a bundle with optional configuration and returns a future.
+   * If completeWhenInstalled is set to true, the future will be completed when the bundle is installed.
+   * Otherwise, the future is completed when the HTTP call to be control protocol is completed.
+   *
+   * @param bundle The file that is the bundle.
+   *               The filename is important with its hex digest string and is required to be consistent
+   *               with the SHA-256 hash of the bundle’s contents.
+   *               Any inconsistency between the hashes will result in the load being rejected.
+   * @param config Optional: Similar in form to the bundle, only that is the file that describes the configuration.
+   *               Again any inconsistency between the hex digest string in the filename, and the SHA-256 digest
+   *               of the actual contents will result in the load being rejected.
+   * @param completeWhenInstalled if set to true, then the future returned will be completed when bundle is installed.
+   *                       This is done by subscribing to bundle events SSE, and checking for bundle installation
+   *                       whenever there are changes in the bundle events.
+   * @param cc implicit connection context
+   * @return The result as a Future[BundleRequestResult]. BundleRequestResult is a sealed trait and can be either:
+   *         - BundleRequestSuccess if the loading request has been succeeded. This object contains the request and bundle id
+   *         - BundleRequestFailure if the loading request has been failed. This object contains the HTTP status code and error message.
+   */
+  def loadBundleComplete(bundleConf: Publisher[Array[Byte]], bundleConfOverlay: Option[Publisher[Array[Byte]]], bundle: BundleFile, config: Option[BundleConfigurationFile], completeWhenInstalled: Boolean = true, completeTimeout: FiniteDuration = 30.seconds)(implicit cc: ConnectionContext): Future[BundleRequestResult] = {
```

Comment:
It shouldn't require Akka as a dependency - it is using reactive streams. Reactive Streams are part of JDK 9. It will be fine to have reactive streams as a dependency of the Scala API.